app/services/accept_npq_application.rb

```diff
@@ -0,0 +1,105 @@
+# frozen_string_literal: true
+
+class AcceptNPQApplication
+  include ActiveModel::Model
+  include ActiveModel::Attributes
+
+  attribute :npq_application
+
+  validates :npq_application, presence: { message: I18n.t(:missing_npq_application) }
+  validate :not_already_accepted
+  validate :cannot_change_from_rejected
+  validate :other_accepted_applications_with_same_course?
+
+  def call
+    return self unless valid?
+
+    ApplicationRecord.transaction do
+      teacher_profile.update!(trn: npq_application.teacher_reference_number) if npq_application.teacher_reference_number_verified?
+      participant_profile
+      npq_application.update(lead_provider_approval_status: "accepted") && other_applications_in_same_cohort.update(lead_provider_approval_status: "rejected")
+      deduplicate_by_trn!
+      npq_application
```

Comment:
can this move outside the transaction?

Reply:
Hi @cwrw tks for your review. Addressed in https://github.com/DFE-Digital/early-careers-framework/pull/2674/commits/8c9fde545669f063b8a66cd2c524248a8025fdb5